.github/workflows/pr-dependencies-check.yml

```
@@ -0,0 +1,211 @@
+name: PR Dependency Check
+
+on:
+  workflow_dispatch:
+    inputs:
+      pr_number:
+        description: 'PR number or URL'
+        type: string
+        required: true
+      search_depth:
+        description: 'Dependencies search depth'
+        required: true
+        default: '100'
+        type: choice
+        options: ['100', '200', '500']
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Pytorch repository
+        uses: malfet/checkout@silent-checkout
+        with:
+          repository: pytorch/pytorch
+          fetch-depth: -1
+          submodules: false
+          quiet-checkout: true
+
+      - name: Set up Git
+        run: |
+          git config --global user.name 'github-actions'
+          git config --global user.email 'github-actions@github.com'
+
+      - name: Check PR dependencies
```

Nit (shell can be sh or something else, and then your script will not work as expected)

Also suggestion/todo: why can't it be a Python script?
```suggestion
      - name: Check PR dependencies
         shell: bash
```

---

> Also suggestion/todo: why can't it be a Python script?

It started as a prototype/proof of concept, the plan is to rewrite it in python and integrate into ghimport at some point.